### PR TITLE
Make tensorflow optional

### PR DESCRIPTION
**Pull request by aireenmei**
# Description

* allow train.py to run in non-tf environment
* For modules, raise clear error message
* For standalone script requiring tf, use try-except block

# Tests
added unit test to ensure importing train.py works without TF 

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/trainers/post_train/distillation/save_top_k_teacher_logits.py [hunk 1]
```diff
@@ -26,7 +26,12 @@
 import time
 import sys
 from etils import epath
-import tensorflow as tf
+
+try:
+  import tensorflow_datasets as tfds  # type: ignore
+except ImportError:
+  sys.exit("tensorflow_datasets is required. Run `pip install tensorflow-datasets`")
```
[the remark this answers is not in the document]
Comment (aireenmei) [reply src/maxtext/trainers/post_train/distillation/save_top_k_teacher_logits.py]: Good catch! fixed

File: tools/data_generation/reshard_tfds.py [hunk 2]
```diff
@@ -49,6 +49,13 @@
 
 import tensorflow as tf
 import tensorflow_datasets as tfds
+
+try:
+  import tensorflow as tf
+  import tensorflow_datasets as tfds
+except ImportError:
+  sys.exit("Tensorflow and tensorflow_datasets are required. Run `pip install tensorflow tensorflow-datasets`")
```
[the remark this answers is not in the document]
Comment (aireenmei) [reply tools/data_generation/reshard_tfds.py:55]: Good catch! fixed

**Review by SurbhiJainUSC — APPROVED** [review]

**Review by NuojCheng — APPROVED** [review]
